tests/pgsmo_tests/utils.py

```diff
@@ -115,20 +118,98 @@ def __init__(
             host: str = 'localhost',
             port: str = '25565',
             user: str = 'postgres'):
+
         # Setup the properties
-        self._server_version = version
+        self._server_version: Tuple[int, int, int] = (
+            int(version[:-4]),
+            int(version[-4:-2]),
+            int(version[-2:])
+        )
+        self.host_name = host
+        self.port = port
+        self.database_name = name
+        self.user_name = user
+        self._provider_name = PG_PROVIDER_NAME
+        self._server_type = "PostgreSQL"
+        self._database_error = DatabaseError
+        self._conn = MockPsycopgConnection({'version': version, 
+                                                'name': name, 
+                                                'host': host, 
+                                                'port': port, 
+                                                'user': user
+                                                }, cur)
+        self.transaction_in_error = False
+        self.notices = []
 
         # Setup mocks for the connection
         self.close = mock.MagicMock()
         self.cursor = mock.MagicMock(return_value=cur)
 
-        dsn_params = {'dbname': name, 'host': host, 'port': port, 'user': user}
-        self.get_dsn_parameters = mock.MagicMock(return_value=dsn_params)
-
     @property
     def server_version(self):
         return self._server_version
+    
+    @property
+    def server_type(self) -> str:
+        """Server type for distinguishing between standard PG and PG supersets"""
+        return self._server_type
+
+    @property
+    def autocommit(self) -> bool:
+        """Returns the current autocommit status for this connection"""
+        pass
+
+    @property
+    def database_error(self):
+        """Returns the type of database error this connection throws"""
+        return self._database_error
+        
+    @property
+    def cancellation_query(self) -> str:
+        backend_pid = self._conn.get_backend_pid()
+        return PG_CANCELLATION_QUERY.format(backend_pid)
+
+    @property
+    def query_canceled_error(self) -> Exception:
+        """Returns driver query canceled error"""
+        return QueryCanceledError
+
+    @autocommit.setter
+    def autocommit(self, mode: bool):
+        """
+        Sets the current autocommit status for this connection
+        :param mode: True or False
+        """
+        pass
+
+    def get_database_owner(self):
+        """
+        List the owner(s) of the current database
+        """
+        database_name = self.database_name
```

Comment:
Why do we need to define these methods in mock class?

Reply:
There are a few places that Sasank changed to use connection.database_name, like in AdminService _handle_get_database_info_request and the Server class __init__ in pgsmo/objects/server,  so I added it here to avoid AttributeErrors.